### PR TITLE
Add more executable comments for Collection

### PR DESCRIPTION
**Pull request by privat**
Fell free to comment

### CODE REVIEW

File: src/Collections-Abstract/Collection.class.st [hunk 1]
```diff
@@ -228,11 +242,13 @@ Collection >> anySatisfy: aBlock [
 
 { #category : #converting }
 Collection >> asArray [
-	"Answer an Array whose elements are the elements of the receiver.
+	"Answer an new Array whose elements are the elements of the receiver.
```
Comment (Ducasse) [inline src/Collections-Abstract/Collection.class.st]: I'm not sure about the new part because if you send #(1 2 3) asArray it probably returns self so just the receiver and not a copy.

File: src/Collections-Abstract/Collection.class.st [hunk 2]
```diff
@@ -288,11 +310,26 @@ Collection >> asCommaStringAnd [
...
+	
+	"|x| x := {'one' -> 1. 'two' ->2} asDictionary. x asDictionary == x >>> true"
+
 	^ self as: Dictionary
 ]
 
 { #category : #converting }
 Collection >> asIdentitySet [
+
+	"Return a new IdemtitySet based on self."
```
Comment (Ducasse) [inline src/Collections-Abstract/Collection.class.st]: Idemtity -> Identity

File: src/Collections-Abstract/Collection.class.st (hunk 1 above)
Comment (privat) [reply src/Collections-Abstract/Collection.class.st]: You are right. The method builds a new array, but is redefined in Array to return self.
So I will assume for now that all the `asX` methods try to return self if applicable.

**Comment by MarcusDenker** [comment]
Failing tests not related:
- [CI] Failing test: ReleaseTest>>testUndeclared #11670
- [CI] Failing test: ProperMethodCategorizationTest>>testNoUncategorizedMethods #11669

**Comment by hogoww** [comment]
I haven't read everything (you did a LOT of them !)
All I read look very good.
One question though, why use bags in the test cases of the collections ? 
Also, you can use #() to create arrays with statics elements in them rather than {} which creates it at run time.

**Comment by privat** [comment]
>  One question though, why use bags in the test cases of the collections ?

I used bags because `asBag` is shorter to write than `asOrderedCollection` and also this shows examples on non sequenceable collections.

>  Also, you can use #() to create arrays with statics elements in them rather than {} which creates it at run time.

From an example point of view, I'm not sure what is the *recommended* usage of Pharoers.

My external point of view is that `#()` is more cumbersome than '{}' since you cannot mutate it or edit the source code to add points or fractions. Basically I see `#()` as a low level optimization, but optimization is not required when writing docs (and premature optimization is the root of all evil bla bla bla)

The only nice thing of `#()` (beside speed and memory efficiency) is that you only need space as separator.
Ok `{1.2.3.4.5.6}` is equally source-code efficient, but some people might find it hard to parse, so it is not something you want in an example.

Multiple times I tried to settle on `#{}` but the parser does not approve my way of making a compromise. :)

**Comment by hogoww** [comment]
All of that is very true !
But executable comments are more than documentation.
There are also unit test cases.

AFAIK They were supposed to be executed in the CI/Test runners, I do not know if they do it currently.

My point is that performance may matter, although probably not **Not that much**

**Comment by Ducasse** [comment]
What we should see is that it would be good to limit dependency between classes.
To me Collection, Array and OrderedCollection are really in the core while we could imagine that Bag can be packaged separately. For example what I envision is a minimal core and that we have well packaged collections that people can load when they want. This is the http://github.com/pharo-containers/ project. 
I have Stack, ...

In the past we did the mistake to "improve Date and Time" while now I would have prefered to clone Date and Time and improves it and reduce the core of Date and Time in Core.
I would like to have a way smaller core and well packaged libraries.

**Comment by privat** [comment]
@Ducasse I thought that Bag was popular, but there is only 49 `Bag` and 10 `asBag`, whereas there is 418 `Set` and 278 `asSet` in my pharo11 image. So I will reroll the PR and restrict myself to more core classes : Collection, Array and OrderedCollection.

I assume it is OK to add `Set`, `Interval` and `String` to the happy few?

**Comment by Ducasse** [comment]
Yes!

**Comment by privat** [comment]
I did a big reroll, and moved things I like less in a distinct PR #11699.
I can split the PR further if it is too much to review at once.

**Comment by privat** [comment]
@Ducasse if you merged #11699 , you should merge this one also
